assets/js/base/components/cart-checkout/shipping-rates-control-package/index.tsx

```diff
@@ -66,13 +68,18 @@ export const ShippingRatesControlPackage = ( {
 	showItems = false,
 }: PackageProps ): ReactElement => {
 	const { selectShippingRate } = useSelectShippingRate();
+	const data = __experimentalApplyCheckoutFilter( {
+		filterName: 'itemName',
+		defaultValue: packageData.name,
+	} );
 
 	const header = (
 		<>
 			{ ( showItems || collapsible ) && (
-				<div className="wc-block-components-shipping-rates-control__package-title">
-					{ packageData.name }
-				</div>
+				<div
+					className="wc-block-components-shipping-rates-control__package-title"
+					dangerouslySetInnerHTML={ { __html: data } }
```

Comment:
Is it possible for this to open up opportunities for XSS attacks? Can you take a look at #7145 and see how the HTML is limited here? Do you think we should we take the same approach?

Reply:
> Is it possible for this to open up opportunities for XSS attacks? 

Nice catch, @opr! I was able to simulate an XSS attack by adding the following code snippet to the site:
```php
add_filter( 'woocommerce_shipping_package_name', 'test_xss', 1, 3 );
function test_xss( $title, $index, $package ) {
    return '<img src=x onerror="&#0000106&#0000097&#0000118&#0000097&#0000115&#0000099&#0000114&#0000105&#0000112&#0000116&#0000058&#0000097&#0000108&#0000101&#0000114&#0000116&#0000040&#0000039&#0000088&#0000083&#0000083&#0000039&#0000041">';
}
```

> Can you take a look at https://github.com/woocommerce/woocommerce-blocks/pull/7145 and see how the HTML is limited here? Do you think we should we take the same approach?

Thanks for pointing this out. I see that the HTML gets sanitized based on a whitelist with HTML tags and HTML attributes. For consistency reasons, I'd use the same logic in this PR.